stack/base/Dockerfile

```diff
@@ -4,6 +4,14 @@ FROM ${BASE}
 LABEL maintainer="AiiDAlab Team <aiidalab@materialscloud.org>"
 
 USER root
+
+RUN apt-get update --yes && \
+    apt-get install --yes --no-install-recommends \
+    # for apps which need to install pymatgen:
+    # https://pymatgen.org/installation.html#installation-tips-for-optional-libraries
+    build-essential && \
```

Comment:
I wonder if the whole `build-essential` package is needed? Did you try installing gcc directly?

Reply:
I did not test but the `build-essential` is what has been moved from original minimal-notebook image. Same suggestion in https://github.com/jupyter/docker-stacks/issues/1385#issuecomment-868379259

Reply:
Also seems when google "ubuntu install gcc", it is recommended to install the `build-essential`. I guess some libraries are all needed and should be installed at the same time.

Reply:
I found something here: https://itslinuxfoss.com/install-gcc-ubuntu-22-04/. They do `sudo apt install gcc-12`

But indeed, by default, they all recommend installing `build-essential`. It contains the "essential tools for development".

Reply:
@unkcpz could you maybe just try installing just gcc, to see how the size would be different?

> we do need to run make

Do we? (I do not know, just curious where it is needed)

Reply:
Anyways, I guess the easiest thing to do here would be to install gcc and try to install numpy/pymatgen. If that fails, probably build-essential is the best option. @unkcpz can you try, please?

Reply:
@unkcpz what was the size difference when you only installed gcc?

Those are just missing header files, so I'd say installing `libc6-dev` is safe and I would go for it if there is a meaningful size difference from fulll `build-essential` package. If the difference is small, let's just install the whole thing.

Reply:
current `full-stack` -> 2.22GB
whole `build-essential` -> 2.43GB
only `gcc` + `libc6-dev` -> 2.38GB

I'd vote to have `build-essential`, the size already increase, so make sense for extra ~50MB for all toolchains.

Reply:
I agree with @unkcpz to go for build-essential 👍